src/lib.ts

```diff
@@ -0,0 +1,180 @@
+import 'zx/globals';
+import xdgAppPaths from "xdg-app-paths";
+import path from "node:path";
+import fs from 'fs/promises'
+import { Auth, Config } from './types.js';
+import { deepmerge } from "deepmerge-ts";
+import login from './login.js';
+import packageJson from '../package.json' assert { type: "json" };
+
+let currentTries = 0;
+
+export const VERSION = packageJson.version;
+export const MAX_TRIES = 3;
+export const BUILD_DIR = 'app';
+export const CONFIG_PATH = xdgAppPaths("webstudio").config();
+export const CONFIG_FILE = path.join(CONFIG_PATH, "config.json");
+export let pm: string | null = null;
+export const supportedBuildTypes = ['remix-app-server', 'express', 'architect', 'flyio', 'netlify', 'vercel', 'cloudflare-pages', 'cloudflare-workers', 'deno'];
+
+const HELP = `Usage:
+    $ webstudio commands [flags...]
+  Commands:
+    login <shared link>             Login to Webstudio with shared link
+    sync <projectId>                Download a project's site data
+    build [projectId]               Build a site
+  Flags:
+    --type, -t                      Build type chosen during build command (default: remix-app-server) 
+                                    (options: ${supportedBuildTypes.join(', ')})
+    --debug                         Enable debug mode
+    --help, -h                      Show this help message
+    --version, -v                   Show the version of this script
+`;
+export const showHelp = () => {
+    console.log(HELP);
+}
+export const detectPackageManager = async () => {
+    const originalverbose = $.verbose;
+    $.verbose = false;
+    try {
+        const pnpm = await $`pnpm --version`.exitCode;
+        const yarn = await $`yarn --version`.exitCode;
+        const npm = await $`npm --version`.exitCode;
+        if (pnpm === 0) {
+            return pm = 'pnpm';
+        }
+        if (yarn === 0) {
+            return pm = 'yarn';
+        }
+        if (npm === 0) {
+            return pm = 'npm';
+        }
+        throw new Error('No package manager found');
+    } catch (e) {
+        throw new Error(e);
+    } finally {
+        $.verbose = originalverbose;
+    }
+
+    $.verbose = originalverbose;
+}
+export const prepareConfigPath = async () => {
+    await $`mkdir -p ${CONFIG_PATH}`
+    await checkConfig();
+}
+export const prepareBuildDir = async () => {
+    await $`rm -rf ${BUILD_DIR}`;
+    await $`mkdir -p ${BUILD_DIR}/app/__generated__`;
+    await $`mkdir -p ${BUILD_DIR}/app/routes`;
+}
+export const checkSiteData = async (args) => {
+    const projectId = args.positionals[1];
+    return await $`ls -1 app/${projectId}.json > /dev/null 2>&1`.exitCode;
+}
+
+export const prepareDefaultRemixConfig = async (type: string) => {
```

Comment:
hmm how can we make the fact that its remix just one template or something ... the cli itself needs to be kinda agnostic ... or maybe we need "plugins" for each framework or something? 

This just looks like a monolithic structure, so this file ends up with every possible framework config all in here

Reply:
Also  lets say a user has a custom framework, how would they do it?

Reply:
We need custom plugins for each framework. The build assets needs to be generated in a different way for each frameworks.